src/formatDate.js

```diff
@@ -8,7 +8,84 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatObject = arrayInObject(fromFormat);
+  const toFormatObject = arrayInObject(toFormat);
+  const arrayOldDate = date.split(fromFormat[3]);
+  const dateObject = arrayDateFromObject(arrayOldDate, fromFormat);
+
+  if (fromFormatObject.yy !== toFormatObject.yy) {
+    if (fromFormatObject.yy.length === 4) {
+      dateObject.yy = dateObject.yy.slice(2);
+    } else {
+      if (+dateObject.yy < 30) {
+        dateObject.yy = '20' + dateObject.yy;
+      } else {
+        dateObject.yy = '19' + dateObject.yy;
+      }
+    }
+  }
```

Comment:
The `dateObject` property should be named 'yyyy' when dealing with a year represented with 4 digits and 'yy' when represented with 2 digits for clarity.